craft_grammar/_processor.py

```diff
@@ -68,6 +68,7 @@ def __init__(
         transformer: Callable[[list[Statement], str, str], str] | None = None,
         valid_platforms: Collection[str] | None = None,
         valid_architectures: Collection[str] | None = None,
+        variant: Variant = Variant.UNKNOWN,
```

Comment:
This is a different strategy than what was explained in the issue, right? In this implementation the application is aware of the variant concept and decides which one to select, right?

It looks like a better interface than determining the variant based on target_arch+platforms, especially since the variant concept is already used to post-process the grammar in the applications.

Reply:
Yeah it's different because I came up with several cases where the application might want to apply it explicitly while still providing both values. So it's a generalisation of what was described in the issue. If you leave it as the default, it'll try to auto-detect, or you can select a grammar.